Samples/WindowsStudio/WindowsStudioSample_WinUI/MainWindow.xaml.cs

```diff
@@ -327,26 +405,37 @@ bool IsCompatibleSubtype(string subtype)
                    || string.Compare(subtype, MediaEncodingSubtypes.Rgb32, true) == 0;
         }
 
-        var formats = m_selectedMediaFrameSource.SupportedFormats.Where(
-                        format => (format.FrameRate.Numerator / format.FrameRate.Denominator) > targetFrameRate
-                                  && IsCompatibleSubtype(format.Subtype))?.OrderBy(
-                                      format => Math.Abs((int)(format.VideoFormat.Width * format.VideoFormat.Height) - (targetWidth * targetHeight)));
-
-        m_selectedFormat = formats.FirstOrDefault();
-
+        m_availableFormats = m_selectedMediaFrameSource.SupportedFormats.Where(
+            format => IsCompatibleSubtype(format.Subtype))?.OrderBy(
+                format => Math.Abs((int)(format.VideoFormat.Width * format.VideoFormat.Height) - (targetWidth * targetHeight))
+                + ((format.FrameRate.Numerator / format.FrameRate.Denominator) - targetFramerate));
+        
+        m_selectedFormat = m_availableFormats.FirstOrDefault();
         if (m_selectedFormat == null)
         {
             throw new Exception($"Could not find a valid frame source format to stream with");
         }
         await m_selectedMediaFrameSource.SetFormatAsync(m_selectedFormat);
 
+        // Update UI with all possible formats available on the stream that can be toggled
+        UIFormatsAvailable.SelectionChanged -= UIFormatsAvailable_SelectionChanged;
+        UIFormatsAvailable.ItemsSource = m_availableFormats.Select(
+            format => $"{format.Subtype}: " +
+                      $"{format.VideoFormat.Width}x{format.VideoFormat.Height}" +
+                      $"@{format.FrameRate.Numerator / format.FrameRate.Denominator}fps");
+        UIFormatsAvailable.SelectedIndex = 0;
+        UIFormatsAvailable.SelectionChanged += UIFormatsAvailable_SelectionChanged;
+
         // Connect the camera to the UI element for previewing the camera
-        m_mediaPlayer = new MediaPlayer();
-        m_mediaPlayer.RealTimePlayback = true;
-        m_mediaPlayer.AutoPlay = true;
+        //if (m_mediaPlayer == null)
```

Comment:
is this `if` commented out on purpose?

Reply:
remnant of a previous debug session, removed now